src/gt4py/analysis/passes.py

```diff
@@ -571,6 +653,15 @@ def accumulate_extents(extents: Sequence[Extent]) -> Extent:
             full_extent |= extent
         return full_extent
 
+    @staticmethod
+    def statements_in_multistages(*multi_stages):
+        """Iterator over statements in the multistages, in the order passed."""
+        for ms in multi_stages:
+            for ij_block in ms.ij_blocks:
+                for int_block in ij_block.interval_blocks:
+                    for stmt_info in int_block.stmts:
+                        yield stmt_info.stmt
```

Comment:
Unfortunately the `FieldDependencyGraphCreator` IR visitor does not work with the infos, so this "hack" (or something like it) is necessary.